### PR TITLE
Feature/add text reading pages

### PR DESCRIPTION
**Pull request by jekuaitk**
* Adds `GivDinStemme` entity
* Adds initial `/read/{collection_id}/{delta}` pages
* Adds initial `/thank-you` page

### CODE REVIEW

File: web/modules/custom/giv_din_stemme/js/giv_din_stemme.js [hunk 1]
```diff
@@ -65,18 +51,42 @@ if (navigator.mediaDevices.getUserMedia) {
     mediaRecorder.onstop = function (e) {
       console.log("Last data to read (after MediaRecorder.stop() called).");
```
Comment (martinyde) [inline web/modules/custom/giv_din_stemme/js/giv_din_stemme.js:52]: Is this a leftover?

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php [hunk 2]
```diff
@@ -164,6 +165,122 @@ public function donate(Request $request): array {
...
+   */
+  private function getCountOfGivDinStemmeByCollectionUuid(string $uuid): int {
+    $result = $this->entityTypeManager->getStorage('gds')->loadByProperties([
+      'collection_id' => $uuid,
+    ]);
+
+    return count($result);
+  }
+
+  private function getRandomText(): Node {
```
Comment (martinyde) [inline web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:265]: Missing comment

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php [hunk 3]
```diff
@@ -164,6 +165,122 @@ public function donate(Request $request): array {
...
+
+    $count  = count($nodes);
+    $keys = array_keys($nodes);
+
+    $randomKey = $keys[rand(0, $count - 1)];
+
+    return $nodes[$randomKey];
+  }
+
+  public function thankYou(Request $request): array {
```
Comment (martinyde) [inline web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:278]: Missing comment

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php [hunk 4]
```diff
@@ -180,46 +297,74 @@ public function show(Request $request): array {
...
+      '#textToRead' => $textToRead,
+      '#totalTexts' => $count,
+      '#nextUrl' => '/read/'. $collectionId . '/' . $delta ,
+      '#attached' => [
+        'library' => ['giv_din_stemme/giv_din_stemme'],
+      ],
+    ];
+  }
+
+  private function handlePost(Request $request, string $collection_id, string $delta): RedirectResponse
```
Comment (martinyde) [inline web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:326]: Missing comment

**Review by martinyde — CHANGES_REQUESTED** [review]
Good job. Consider moving some of the controller methods into helper.

File: web/modules/custom/giv_din_stemme/js/giv_din_stemme.js (hunk 1 above)
Comment (jekuaitk) [reply web/modules/custom/giv_din_stemme/js/giv_din_stemme.js:52]: Yes, i will remove all debugging just before merging.

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php (hunk 2 above)
Comment (jekuaitk) [reply web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:265]: Added.

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php (hunk 3 above)
Comment (jekuaitk) [reply web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:278]: Added.

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php [hunk 5]
```diff
@@ -164,6 +165,122 @@ public function donate(Request $request): array {
...
+    $result = $this->entityTypeManager->getStorage('gds')->loadByProperties([
+      'collection_id' => $uuid,
+    ]);
+
+    return count($result);
+  }
+
+  private function getRandomText(): Node {
+
+    $nids = \Drupal::entityQuery('node')->condition('type','text')->accessCheck(FALSE)->execute();
```
[the remark this answers is not in the document]
Comment (jekuaitk) [reply web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:267]: Used injection and moved to helper.

File: web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php (hunk 4 above)
Comment (jekuaitk) [reply web/modules/custom/giv_din_stemme/src/Controller/GivDinStemmeController.php:326]: Added.

**Pull request merged**